src/qudi/hardware/fast_adc/spectrum/si_main.py

```diff
@@ -0,0 +1,411 @@
+# -*- coding: utf-8 -*-
+"""
+This file contains the Qudi hardware for spectrum instrumentation ADC.
+
+Copyright (c) 2021, the qudi developers. See the AUTHORS.md file at the top-level directory of this
+distribution and on <https://github.com/Ulm-IQO/qudi-iqo-modules/>
+
+This file is part of qudi.
+
+Qudi is free software: you can redistribute it and/or modify it under the terms of
+the GNU Lesser General Public License as published by the Free Software Foundation,
+either version 3 of the License, or (at your option) any later version.
+
+Qudi is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with qudi.
+If not, see <https://www.gnu.org/licenses/>.
+"""
+import time
+import numpy as np
+
+from enum import IntEnum
+from pyspcm import *
+from spcm_tools import *
```

Comment:
excplicitly import only the necessary classes and functions, not the whole module.